electricpy/__init__.py

```diff
@@ -4485,42 +4682,95 @@ def characterz(R,G,L,C,freq=60):
                 Capacitance in Farads.
     freq:       float, optional
                 System frequency in Hz, default=60
-    
+
     Returns
     -------
     Zc:         complex
                 Charcteristic Impedance of specified line.
     """
     # Evaluate omega
-    w = 2*_np.pi*freq
+    w = 2 * _np.pi * freq
     # Evaluate Zc
-    Zc = _np.sqrt((R+1j*w*L)/(G+1j*w*C))
-    return(Zc)
+    Zc = _np.sqrt((R + 1j * w * L) / (G + 1j * w * C))
+    return (Zc)
+
+#define propagation_constants for long transmission line
+def propagation_constants(z, y, length):
+    
+    '''
+        this function calculates the propagation constants of long transmission line 
+        
+        long transmission line is governed by differential equation
+        
+        (d^2)V/d(x^2) = (gama)*V
+        
+        Parameters
+        ----------
+        z: impedence of the transmission line R+j*2*pi*f*L
+        
+        y: admitance of the transmission line g+j*2*pi*f*C
+        
+        Returns
+        -------
+        
+        gama: propagation constant
+        
+        zc: surge_impedence
+        
+        alpha : attenuation constant
+        
+        beta: imaginary part of gama
+        
+    
+    '''
+
+    assert length > 500, "long transmission line length should be grater than 500km"
+
+    gama = np.sqrt(z * y)
+
+    alpha = gama.real
+
+    beta = gama.imag
+
+    zc = np.sqrt(z / y)
+
+    params = {
+        'gama': gama,
```

Comment:
"gamma" is the appropriate spelling of the Greek letter, we should probably update this throughout for consistency.

Reply:
this was completed in subsequent PR